File: yurongyang/1137.cpp
class Solution {
public:
    int tmp[38]={0};
    int tribonacci(int n) {
        if(n==0 || n==1 || n==2){
            return n==0?0:1;
        }
        if(n>2 && tmp[n] !=0){
            return tmp[n];
        }
        if(n<=37){
            tmp[n] = tribonacci(n-1)+tribonacci(n-2)+tribonacci(n-3);
            return tmp[n];
        }
        return 0;
    }
};

File: sparrowsong/1137.go
package leetcode

var m = map[int]int{
	0: 0,
	1: 1,
	2: 1,
}

func tribonacci(n int) int {
	if v, ok := m[n]; ok {
		return v
	}
	m[n] = tribonacci(n-1) + tribonacci(n-2) + tribonacci(n-3)
	return m[n]
}
